src/stringify.ts

```diff
@@ -17,78 +30,92 @@ const TYPE_FOR_TO_STRING = new Map([
   ['[object RegExp]', 'RegExp'],
   ['[object Boolean]', 'Boolean'],
   ['[object Null]', 'Null'],
-  ['[object Undefined]', 'Undefined']
+  ['[object Undefined]', 'Undefined'],
 ]);
 
-function detectType(value) {
+function detectType(value: BSONValue) {
   return TYPE_FOR_TO_STRING.get(Object.prototype.toString.call(value));
 }
 
-function getTypeDescriptorForValue(value) {
+function getTypeDescriptorForValue(value: BSONValue) {
   const t = detectType(value);
   const _bsontype = t === 'Object' && value._bsontype;
   return {
     type: _bsontype || t,
-    isBSON: !!_bsontype
+    isBSON: !!_bsontype,
   };
 }
 
 const BSON_TO_JS_STRING = {
-  Code: function(v) {
+  Code: function (v: Code) {
     if (v.scope) {
       return `Code('${v.code}',${JSON.stringify(v.scope)})`;
     }
     return `Code('${v.code}')`;
   },
-  ObjectID: function(v) {
+  ObjectID: function (v: ObjectId) {
     return `ObjectId('${v.toString('hex')}')`;
   },
-  ObjectId: function(v) {
+  ObjectId: function (v: ObjectId) {
     return `ObjectId('${v.toString('hex')}')`;
   },
-  Binary: function(v) {
+  Binary: function (v: Binary) {
     const subType = v.sub_type;
     if (subType === 4 && v.buffer.length === 16) {
-      const uuidHex = v.buffer.toString('hex');
-      return `UUID("${uuidHex.slice(0, 8)}-${uuidHex.slice(8, 12)}-${uuidHex.slice(12, 16)}-${uuidHex.slice(16, 20)}-${uuidHex.slice(20, 32)}")`;
+      // The `Binary.buffer.toString` type says it doesn't accept
+      // arguments. However it does, and a test will fail without it.
+      const uuidHex = (v.buffer.toString as unknown as Binary['toString'])(
+        'hex'
+      );
+      return `UUID("${uuidHex.slice(0, 8)}-${uuidHex.slice(
+        8,
+        12
+      )}-${uuidHex.slice(12, 16)}-${uuidHex.slice(16, 20)}-${uuidHex.slice(
+        20,
+        32
+      )}")`;
     }
-    return `BinData(${subType.toString(16)}, '${v.buffer.toString('base64')}')`;
+    // The `Binary.buffer.toString` type says it doesn't accept
+    // arguments. However it does, and a test will fail without it.
+    return `BinData(${subType.toString(16)}, '${(
+      v.buffer.toString as unknown as Binary['toString']
+    )('base64')}')`;
```

Comment:
You can use toString right on the Binary, you don't need to do it on the buffer directly

```suggestion
    return `BinData(${subType.toString(16)}, '${v.toString('base64')}')`;
```